### ISSUE TITLE
Unclear inference result

### ISSUE THREAD

**Issue by junkeon** [janvainer/speedyspeech#3]
Hi, I trained duration extractor and speedyspeech with my own dataset which is korean single speech dataset
(RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 22050 Hz)
I just changed dataloader part for my dataset but everything except that use defualt setting and i also use pwgan instead of melgan
As i attached the wav file, the result is not what i expect
Start of speech is clear enough but is getting unclear
Can you please give an advice for me?
[lmy.zip](https://github.com/janvainer/speedyspeech/files/5139507/lmy.zip)

**Comment by LordOfLuck** [comment]
Hi, thanks for trying out the model! The audio data format seems to be ok. The synthesized audio was created with the student model? Could you please also send some reference audio of the synthesized one? (I unfortunately do not speak korean so it is a bit more difficult to judge the synthesis quality). Regarding the dataloader changes, whatd did you use to preprocess the text? Did you use phonemes to train the models?

**Comment by janvainer** [comment]
Hi, thanks for trying out the model! The audio data format seems to be ok. The synthesized audio was created with the student model? Could you please also send some reference audio of the synthesized one? (I unfortunately do not speak korean so it is a bit more difficult to judge the synthesis quality). Regarding the dataloader changes, whatd did you use to preprocess the text? Did you use phonemes to train the models?

**Comment by junkeon** [comment]
[speedyspeech_custom.zip](https://github.com/janvainer/speedyspeech/files/5141265/speedyspeech_custom.zip)
Thanks for your kind reply :)

I attached the files
The files with name 'gt' are vocoded files with ground truth mel and the files with name 'ss' are inferenced files with trained speedyspeech. (I hope these gt files help you)
As you can see, the first part seems clear but it is getting unclear and mumble (what i feel... i am sorry, i don't know how to explain this...)
I used g2p module for korean, so every characters are transfered to phoneme. 
I am pretty sure there is no problem in g2p module becuase i have been used the module and it workes well in other model.

I wonder there were any similar case while you train speech data.
I will keep searching and finding some missing part.

Thank you! i really love your model! it's awesome!!

**Comment by janvainer** [comment]
Hm I see now. I am not able to exactly point to a specific issue but I generally encountered the following problems during my experiments:
1. The duration extractor needs to be trained to convergence. The student model suffers a lot if the durations are low quality. Maybe try trining the duration extractor a bit longer or if you have some pretrained tacotron 2 for korean available, you could try to use the durations from there.

**Comment by junkeon** [comment]
Thanks for your reply and sorry for late comment
As your advice, I need to check duration extractor quality and preprocessing.
i will check them and leave comments.

Thank you!

**Comment by junkeon** [comment]
![image](https://user-images.githubusercontent.com/35945268/91697143-9631c300-ebab-11ea-9542-5b4832c02b96.png)

As you said, duration extractor (teacher model) is not working and that is why student model inference result seems wierd

i will check duration extractor first

Thank you!!!

**Comment by junkeon** [comment]
I got clear mel spectrogram from both duration extractor and speedyspeech and it seems fine
The problem i mentioned was caused by the scale difference between sigmoid and my mel

![image](https://user-images.githubusercontent.com/35945268/91784475-fcffbc80-ec3d-11ea-9dff-961f5a8a18e6.png)

![image](https://user-images.githubusercontent.com/35945268/91784521-17399a80-ec3e-11ea-8366-f61c31852d81.png)



Thank you so much for helping me!

**Comment by LordOfLuck** [comment]
@junkeon Great! thanks for letting me know ;) I was wondering, what vocoding did you use for your experiments? Could you provide some link to paper or github repo? The result seems quite clean.

**Comment by junkeon** [comment]
oh sure!
I use Parallel wavegan model for vocoding, PWGAN is the best among our candidates.

Paper : https://arxiv.org/abs/1910.11480
Code : https://github.com/kan-bayashi/ParallelWaveGAN

**Comment by LordOfLuck** [comment]
Thanks! I havent read this one yet :). Let me know if you get any more success or trouble with SpeedySpeech.

**Comment by hylee-250** [comment]
@junkeon 안녕하세요, speedyspeech 모델을 KSS 데이터셋으로 학습시키고 싶은데 한국어 데이터셋 전처리 하는 방법을 몰라서 utils/text.py의 TextProcessor class를 변형해서 G2PK를 적용하려고 하는데 영어는 Phoneme 목록을 먼저 선언하고 인덱스를 가져올 수 있지만 한글은 초성, 중성, 종성으로 나뉘어서 방법이 조금 다를 것 같은데 혹시 한국어 전처리를 어떻게 사용하셨는지 알수 있을까요?
